Sources/Main.cpp

```diff
@@ -758,7 +850,7 @@ void __fastcall TX584Form::InputEditExit(TObject *Sender)
 {
     //завершаем редактирование и перерисовываем строку
     if (InputEdit->Visible) {
-        CodeListView->Items->Item[EditRow]->SubItems->Strings[2] = InputEdit->Text;
+        CodeListView->Items->Item[EditRow]->SubItems->Strings[EditColumn] = InputEdit->Text;
```

Comment:
What if the user enters an invalid control flow statement? Previously we silently ignored ill-formed statements and considered them as ordinary comments. Since we added a dedicated column for control flow statements, we can't ignore syntax errors anymore.

I would suggest to check the contents of `InputEdit` via `ParseComment`. If the user made a mistake, show an error dialog and cancel the editing. For now, it is enough to show a generic error message about syntax error in control flow statement. Later we can implement a detailed error reporting and tell the user what exactly went wrong. Would this work?

Reply:
Thank you for proposition, I think that this will prevent typos and mistakes